File: Lab 6/Strings/MyString.cs
namespace Strings
{
    abstract public class MyString
    {
        public MyString()
        {

        }
        public MyString(char[] str)
        {
            length = str.Length;
            this.str = new char[length];
            for (int i = 0; i < length; i++)
            {
                this.str[i] = str[i];
            }
        }
        abstract public MyString SubtractSymbol();
        public virtual int GetLength()
        {
            return length;
        }
        public char[] GetLine()
        {
            return str;
        }
        protected char[] str;
        protected int length;

    }
}

File: Lab 9/C++/MyString.cpp
#include "MyString.h"

MyString::MyString(const char* str)
{
	length = 0;
	while (str[length] != '\0')
		length++;
	this->str = new char[length + 1];
	for (int i = 0; i < length; i++) {
		this->str[i] = str[i];
	}
	this->str[length] = '\0';
}

MyString::~MyString()
{
	delete[]str;
}

int MyString::GetLength()
{
	return length;
}

char* MyString::GetLine()
{
	return str;
}
